docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/FeatureSelection/SelectFeaturesBasedOnMutualInformation.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.ML;
+using Microsoft.ML.Data;
+
+namespace Samples.Dynamic
+{
+    public static class SelectFeaturesBasedOnMutualInformation
+    {
+        public static void Example()
+        {
+            // Create a new ML context, for ML.NET operations. It can be used for exception tracking and logging, 
+            // as well as the source of randomness.
+            var mlContext = new MLContext();
+
+            // Get a small dataset as an IEnumerable and convert it to an IDataView.
+            var rawData = GetData();
+
+            Console.WriteLine("Contents of two columns 'Label' and 'NumericVector'.");
+            foreach (var item in rawData)
+                Console.WriteLine("{0}\t\t{1}", item.Label, string.Join(" ", item.NumericVector));
+            // True            4 0 6
+            // False           0 5 7
+            // True            4 0 6
+            // False           0 5 7
+
+            var data = mlContext.Data.LoadFromEnumerable(rawData);
+
+            // We define a MutualInformationFeatureSelectingEstimator that selects the top k slots in a feature 
+            // vector based on highest mutual information between that slot and a specified label. 
+
+            var pipeline = mlContext.Transforms.FeatureSelection.SelectFeaturesBasedOnMutualInformation(
+                outputColumnName: "NumericVector", labelColumnName: "Label",
+                slotsInOutput:2);
+
+            // The pipeline can then be trained, using .Fit(), and the resulting transformer can be used to transform data. 
+            var transformedData = pipeline.Fit(data).Transform(data);
+
+            Console.WriteLine("Contents of column 'NumericVector'");
+            PrintDataColumn(transformedData, "NumericVector");
+            // 4 0
```

Comment:
>4 [](start = 15, length = 1)

Curious, why is it dropping 6, but keeping 4? It is not obvious to me.  A comment about that might help.

Reply:
we need to improve this once we have found a better sample. i have noted this in the issue #1209

---
In reply to: [272397744](https://github.com/dotnet/machinelearning/pull/3184#discussion_r272397744) [](ancestors = 272397744)